cpu/esp8266/ld/esp8266.riot-os.ld

```diff
@@ -286,7 +286,7 @@ SECTIONS
     *libc.a:*fputwc.o(.literal .text .literal.* .text.*)
     */
 
-    enc28j60.a:*(.literal .text .literal.* .text.*)
+    *enc28j60/*(.literal .text .literal.* .text.*)
```

Comment:
What's with this random driver getting special treatment :thinking:

Reply:
I'm not really sure, maybe @gschorcht can clarify?

Reply:
Hm, yes I can clarify. In section [Network Devices](https://doc.riot-os.org/group__cpu__esp8266.html#esp8266_network_devices) of the ESP8266 documentation it is mentioned that ESP8266 is working and has been tested with ENC28J60 and MRF24J40. However, the ENC28J60 works only if the driver code is placed in IRAM.

Since the IRAM of the ESP8266 is very limited and only has a size of 48 kByte, it is not possible to place code segments in the IRAM by default. The IROM of the ESP8266 on the other hand has 448 kByte. The IRAM is of course much faster, while the IROM is much slower and can only be accessed via a cache, which is also sometimes disabled, e.g. by the WiFi module or when writing to the flash.

Therefore, time-critical code as well as code that has to work even when the cache is disabled must be placed in the IRAM. Due to the limited size of the IRAM, fine-tuning is always required as to what must be placed in the IRAM and what can be placed in the IROM.

Reply:
> IROM is much slower and can only be accessed via a cache, which is also sometimes disabled, e.g. by the WiFi module

The WiFi module disabling the cache is surprising! Wouldn't that mean that *any* code can be hit by the problem if another task gets scheduled while the WiFi driver has the cache disabled? Or are interrupts disabled together with the cache?

Reply:
@benpicco I'm not really sure (I just forgot too much in last 2 years doing other things :worried:), but when I remember correctly, I had problems to access IROM when the WiFi module was active. So my guess at that time was (without knowing it definitely, there is no documentation) that it might be that the WiFi module is connected via SPI inside the SoC. That's why `esp_wifi` is already in IRAM. I could try it again.